_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -1492,6 +1493,7 @@ mfxStatus ImplementationAvc::Reset(mfxVideoParam *par)
     }
     m_video = newPar;
 
+    InitDTSList();
```

Comment:
Reset function can be called for on-the-fly parameters change (e.g. bitrate, framerate, number of slices, etc.). In this case internal encoder state isn't reset. I.e. counters are not dropped, all the calculations (GOP generation, HRD parameters generation, etc.) are smoothly continued for next frames after Reset. This relates to DTS either. Please move call of "InitDTSList()" to the code under "if (isIdrRequired || IsOn(extResetOpt->StartNewSequence))" - this code does deep encoder reset.